src/app/cluster/cluster-details/alertmanager-config/alertmanager-config-dialog/component.spec.ts

```diff
@@ -0,0 +1,104 @@
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
+import {fakeAlertmanagerConfig} from '@app/testing/fake-data/mla';
+import {asyncData} from '@app/testing/services/api-mock';
+import {MatDialogRefMock} from '@app/testing/services/mat-dialog-ref-mock';
+import {CoreModule} from '@core/module';
+import {NotificationService} from '@core/services/notification';
+import {MLAService} from '@core/services/mla';
+import {SharedModule} from '@shared/module';
+import {NGX_MONACO_EDITOR_CONFIG, MonacoEditorModule} from 'ngx-monaco-editor';
+import {AlertmanagerConfigDialog, Mode} from './component';
+
+const modules: any[] = [BrowserModule, BrowserAnimationsModule, SharedModule, CoreModule, MonacoEditorModule];
+
+declare let monaco: any;
+
+describe('AlertmanagerConfigDialog', () => {
+  let fixture: ComponentFixture<AlertmanagerConfigDialog>;
+  let component: AlertmanagerConfigDialog;
+  let putAlertmanagerConfigSpy;
+
+  beforeEach(
+    waitForAsync(() => {
+      const mlaMock = {
+        putAlertmanagerConfig: jest.fn(),
+        refreshAlertmanagerConfig: () => {},
+      };
+      putAlertmanagerConfigSpy = mlaMock.putAlertmanagerConfig.mockReturnValue(asyncData(fakeAlertmanagerConfig()));
+
+      TestBed.configureTestingModule({
+        imports: [...modules],
+        declarations: [AlertmanagerConfigDialog],
+        providers: [
+          {provide: MatDialogRef, useClass: MatDialogRefMock},
+          {provide: MLAService, useValue: mlaMock},
+          {
+            provide: MAT_DIALOG_DATA,
+            useValue: {
+              title: '',
+              projectId: '',
+              clusterId: '',
+              mode: '',
+              confirmLabel: '',
+            },
+          },
+          NotificationService,
+          {provide: NGX_MONACO_EDITOR_CONFIG, useValue: {onMonacoLoad: () => (monaco = (window as any).monaco)}},
+        ],
+      }).compileComponents();
+    })
+  );
+
+  describe('Edit Alertmanager Config Dialog', () => {
+    beforeEach(() => {
+      fixture = TestBed.createComponent(AlertmanagerConfigDialog);
+      component = fixture.componentInstance;
+      fixture.detectChanges();
+
+      component.data = {
+        title: 'Edit Alertmanager Config',
+        projectId: '123ab4cd5e',
+        clusterId: '4k6txp5sq',
+        mode: Mode.Edit,
+        confirmLabel: 'Edit',
+        alertmanagerConfig: fakeAlertmanagerConfig(),
+      };
+      fixture.detectChanges();
+    });
+
+    it(
+      'should create the edit alertmanager config dialog',
+      waitForAsync(() => {
+        expect(component).toBeTruthy();
+      })
+    );
+
+    it('should have correct title: edit', () => {
+      expect(document.body.querySelector('km-dialog-title').textContent).toContain('Edit Alertmanager Config');
+    });
+
+    it('should have correct button text: edit', () => {
+      expect(document.body.querySelector('#km-alertmanager-config-dialog-btn').textContent).toContain(' Edit ');
```

Comment:
Why there are spaces around `Edit`?

Reply:
This linebreak is causing the extra spaces. We have the same "problem" on all buttons due to that.
```
  <button id="km-alertmanager-config-dialog-btn"
          mat-flat-button
          (click)="save()"
          [disabled]="!isValid()">
    {{data.confirmLabel}}
  </button>
```

Reply:
You could also do a trim on the text before comparing.

Reply:
I didn't knew that. `trim()` sounds good.

Reply:
BTW, the method that is used in the test is `toContain` perhaps it would work without spaces?

```
Used when you want to check that an item is in a list. For testing the items in the list, this uses ===, a strict equality check. It can also check whether a string is a substring of another string.
```

Reply:
But that's just a minor thing, don't spend too much time on it. We can keep it like this if it takes a lot of time.

Reply:
@maciaszczykm you are right `toContain` does the trick itself. I remembered some failures in the past due to missing whitespaces, but it seems it was not a problem with `toContain`. 👍 Changed.